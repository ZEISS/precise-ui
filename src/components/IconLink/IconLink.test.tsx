import * as React from 'react';
import * as enzyme from 'enzyme';
import { IconLink } from './';

const defaultIconName = 'Favorite';

describe('<IconLink />', () => {
  it('should render an empty <IconLink> component', () => {
    const wrapper = enzyme.shallow(<IconLink icon={defaultIconName} />);
    expect(wrapper.text()).toBe('<Styled(Anchor) />');
  });

  it('should render a <IconLink> component with some content', () => {
    const wrapper = enzyme.mount(<IconLink icon={defaultIconName}>Content</IconLink>).mount();
    expect(wrapper.text()).toBe('Content');
  });

  it('should render a disabled <IconLink> with some content', () => {
    const wrapper = enzyme.shallow(
      <IconLink icon={defaultIconName} disabled>
        Content
      </IconLink>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <IconLink> with some content component', () => {
    const wrapper = enzyme.shallow(
      <IconLink icon={defaultIconName} href="foo">
        <div>Content</div>
      </IconLink>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should execute the click handler if enabled', () => {
    let clicked = false;
    const clickHandler = () => {
      clicked = true;
    };
    const wrapper = enzyme.mount(
      <IconLink icon={defaultIconName} onClick={clickHandler}>
        Content
      </IconLink>,
    );
    wrapper.find('a').simulate('click', {
      preventDefault() {},
    });
    expect(clicked).toBe(true);
  });

  it('should not execute the click handler if disabled', () => {
    let clicked = false;
    const clickHandler = () => {
      clicked = true;
    };
    const wrapper = enzyme.mount(
      <IconLink icon={defaultIconName} onClick={clickHandler} disabled>
        Content
      </IconLink>,
    );
    wrapper.find('a').simulate('click', {
      preventDefault() {},
    });
    expect(clicked).toBe(false);
  });

  it('should not fail if no history given', () => {
    let clicked = false;
    const clickHandler = () => {
      clicked = true;
    };
    const wrapper = enzyme.mount(
      <IconLink icon={defaultIconName} to="/foo" onClick={clickHandler}>
        Content
      </IconLink>,
    );
    wrapper.find('a').simulate('click', {
      preventDefault() {},
    });
    expect(clicked).toBe(true);
  });
});
