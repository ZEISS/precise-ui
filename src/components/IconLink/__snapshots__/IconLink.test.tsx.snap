// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconLink /> should render a <IconLink> with some content component 1`] = `
<Styled(Anchor)
  display="inline-block"
  href="foo"
>
  <Styled(Icon)
    iconOnly={false}
    name="Favorite"
    size="22px"
  />
  <styled.span>
    <div>
      Content
    </div>
  </styled.span>
</Styled(Anchor)>
`;

exports[`<IconLink /> should render a disabled <IconLink> with some content 1`] = `
<Styled(Anchor)
  disabled={true}
  display="inline-block"
>
  <Styled(Icon)
    disabled={true}
    iconOnly={false}
    name="Favorite"
    size="22px"
  />
  <styled.span>
    Content
  </styled.span>
</Styled(Anchor)>
`;
